main.tf

```diff
@@ -0,0 +1,55 @@
+provider "aws" {
+  alias = "glue"
+}
+
+data "aws_region" "current" {}
+
+data "aws_iam_policy_document" "default" {
+  statement {
+    actions = [
+      "sts:AssumeRole"
+    ]
+    principals {
+      type        = "Service"
+      identifiers = ["glue.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "default" {
+  count              = var.role_arn == null ? 1 : 0
+  name               = "GlueRole-${var.name}"
+  assume_role_policy = data.aws_iam_policy_document.default.json
+  tags               = var.tags
+}
+
+resource "aws_iam_role_policy" "default" {
+  count  = var.role_arn == null ? 1 : 0
+  name   = "GlueRole-${var.name}"
+  role   = aws_iam_role.default[0].id
+  policy = var.policy
+}
+
+resource "aws_iam_role_policy_attachment" "default" {
+  provider   = aws.glue
+  count      = var.role_arn == null
+  role       = aws_iam_role.default[0].id
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole"
+}
```

Comment:
Maybe use the MCAF role module instead?

Reply:
Can i use count in the MCAF role module?

Reply:
Ah, I see... You only create the role if no `role_arn` is given. In that case I think this is fine. But you need to fix line 35 as it misses the `? 1 : 0` part